src/backing/client/crypto_key_version_algorithm_test.cc

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "google/cloud/kms/v1/resources.pb.h"
+#include "src/backing/client/crypto_key_version_algorithm.h"
+
+namespace kmsengine {
+namespace backing {
+namespace {
+
+using ::testing::ValuesIn;
+
+using CryptoKeyVersion = google::cloud::kms::v1::CryptoKeyVersion;
+
+// Mapping between `DigestCase` cases and their protobuf equivalents.
+struct CorrespondingCryptoKeyVersionAlgorithm {
+  CryptoKeyVersionAlgorithm actual;
+  google::cloud::kms::v1::CryptoKeyVersion_CryptoKeyVersionAlgorithm proto;
+} const kCryptoKeyVersionAlgorithmMapping[]{
```

Comment:
Newline?

Reply:
There's no newline here because the type of the const is the struct itself. Could define it to be:

```
struct CorrespondingCryptoKeyVersionAlgorithm {
  CryptoKeyVersionAlgorithm actual;
  google::cloud::kms::v1::CryptoKeyVersion_CryptoKeyVersionAlgorithm proto;
};

const CorrespondingCryptoKeyVersionAlgorithm kCryptoKeyVersionAlgorithmMapping[] { ... };
```

instead, if that's preferred

Reply:
Ah, I see. I think I'd mildly prefer the alternative. Defining the struct type in the same statement where a variable of that type is declared is not something you see often.